packages/atomic/src/components/commerce/atomic-commerce-products-per-page/atomic-commerce-products-per-page.spec.ts

```diff
@@ -0,0 +1,348 @@
+        template: html`<atomic-commerce-products-per-page
+          choices-displayed=${ifDefined(props.choicesDisplayed)}
+          initial-choice=${ifDefined(props.initialChoice)}
+        ></atomic-commerce-products-per-page>`,
+        selector: 'atomic-commerce-products-per-page',
+        bindings: (bindings) => {
+          bindings.interfaceElement.type = interfaceElementType;
+          bindings.engine = mockedEngine;
+          bindings.store.state.resultList = {
+            focusOnFirstResultAfterNextSearch:
+              mockedFocusOnFirstResultAfterNextSearch,
+            focusOnNextNewResult: mockedFocusOnNextNewResult,
+          };
+          return bindings;
+        },
+      });
+
+    return {
+      element,
+      get legend() {
+        return element.shadowRoot!.querySelector('legend');
+      },
+      get label() {
+        return element.shadowRoot!.querySelector('span[part="label"]');
+      },
+      get buttonList() {
+        return element.shadowRoot!.querySelector('div[part="buttons"]');
+      },
+      get buttons() {
+        return element.shadowRoot!.querySelectorAll('input[part*="button"]');
+      },
+      get activeButton() {
+        return element.shadowRoot!.querySelector(
+          'input[part*="active-button"]'
+        );
+      },
+    };
+  };
+
+  beforeEach(() => {
+    vi.spyOn(console, 'error').mockImplementation(() => {});
+  });
+
+  it('should log an error and throw if choicesDisplayed is not a valid comma-separated list of numbers', async () => {
+    const choicesDisplayed = '10,25,invalid,100';
+
+    const {element} = await renderProductsPerPage({
+      props: {choicesDisplayed},
+    });
+
+    expect(element.error).toBeDefined();
+    expect(element.error.message).toBe(
+      'The choice value "invalid" from the "choicesDisplayed" option is not a number.'
+    );
+  });
+
+  it('should log an error and throw if initialChoice is not in choicesDisplayed', async () => {
+    const choicesDisplayed = '10,25,50,100';
+    const initialChoice = 200;
+
+    const {element} = await renderProductsPerPage({
+      props: {choicesDisplayed, initialChoice},
+    });
+
+    expect(element.error).toBeDefined();
+    expect(element.error.message).toBe(
+      'The initial choice value "200" is not included in the choices 10,25,50,100.'
+    );
+  });
+
+  it('should call buildSearch when interfaceElement type is "search"', async () => {
+    await renderProductsPerPage({
+      interfaceElementType: 'search',
+    });
+
+    expect(buildSearch).toHaveBeenCalledWith(mockedEngine);
+  });
+
+  it('should call buildProductListing when interfaceElement type is "product-listing"', async () => {
+    await renderProductsPerPage({
+      interfaceElementType: 'product-listing',
+    });
+
+    expect(buildProductListing).toHaveBeenCalledWith(mockedEngine);
+  });
+
+  it('should call summary on this.summary', async () => {
```

Comment:
I'm not a fan of this description and the following one.

Suggestion:

"should set this.summary to the product listing or search controller's summary"

similar for pagination.